cvat/apps/engine/static/engine/js/frameClipper.js

```diff
@@ -0,0 +1,20 @@
+class FrameClipper {
+    constructor() {
+        this._clipToFrame = false;
+    }
+
+    clamp(value, min, max) {
+        return this._clipToFrame ? value : Math.clamp(value, min, max);
+    }
+
+    get clipToFrame() {
+        return this._clipToFrame;
+    }
+
+    set clipToFrame(value) {
+        return this._clipToFrame = value;
+    }
+
+}
+
+window.frameClipper = new FrameClipper();
```

Comment:
I had to use global variable, otherwise I would have to add extra parameter to two many functions/constructors.

Reply:
add it to another existing global at appropriate level, e.g. `window.cvat.frameClipper`